lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -26,28 +35,150 @@ impl<TInvertedIndex: InvertedIndex> SparseVectorIndex<TInvertedIndex> {
         id_tracker: Arc<AtomicRefCell<IdTrackerSS>>,
         vector_storage: Arc<AtomicRefCell<VectorStorageEnum>>,
         payload_index: Arc<AtomicRefCell<StructPayloadIndex>>,
-        path: PathBuf,
+        path: &Path, // TODO(sparse) use path to load/save index
         inverted_index: TInvertedIndex,
     ) -> Self {
+        let searches_telemetry = SparseSearchesTelemetry::new();
+        let path = path.to_path_buf();
         Self {
             id_tracker,
             vector_storage,
             payload_index,
             path,
             inverted_index,
+            searches_telemetry,
         }
     }
 
     /// Search index using sparse vector query
     pub fn search_sparse(
         &self,
-        query: SparseVector,
+        vectors: &[&QueryVector],
         top: usize,
         is_stopped: &AtomicBool,
-    ) -> Vec<ScoredPointOffset> {
-        let mut search_context = SearchContext::new(query, top, &self.inverted_index, is_stopped);
-        search_context.search()
+        with_filter: bool,
+        condition: impl Fn(PointOffsetType) -> bool,
+    ) -> OperationResult<Vec<Vec<ScoredPointOffset>>> {
+        let mut result = vec![];
+
+        for vector in vectors {
+            check_process_stopped(is_stopped)?;
+            // measure time according to filter
+            if with_filter {
```

Comment:
`_timer` has drop with timer stopping. we have to move `_timer` to higher scope:
```
            let _timer = if with_filter {
                ScopeDurationMeasurer::new(&self.searches_telemetry.filtered_sparse)
            } else {
                ScopeDurationMeasurer::new(&self.searches_telemetry.unfiltered_sparse)
            };
```